EmpireLegends/GameStart.cpp

```diff
@@ -92,3 +92,30 @@ void GameStart::detPlayerCount() {
 
 	std::cout << "\nEach player gets " << playerCoins << " coins." << std::endl;
 }
+
+void GameStart::startGame()
+{
+	state = "WELCOME TO EMPIRE LEGENDS. THE GAME IS ABOUT TO START.";
+	Notify();
+}
+
+GameStartObserver::GameStartObserver(GameStart* s)
+{
+	subject = s;
+	subject->Attach(this);
+}
+
+GameStartObserver::~GameStartObserver()
+{
+	subject->Detach(this);
```

Comment:
shouldn't detaching observers be the subject's responsibility and not the observer's? That way each subject just detaches their observers all at once when they're destroyed